src/wazuh_modules/wm_sca.c

```diff
@@ -1504,24 +1504,41 @@ static int wm_sca_read_command(char *command, char *pattern,wm_sca_t * data, cha
     char *cmd_output = NULL;
     int result_code;
 
-    if( wm_exec(command,&cmd_output,&result_code,data->commands_timeout,NULL) < 0 )  {
-        if (*reason == NULL){
-            os_malloc(OS_MAXSTR, *reason);
-            if (result_code == EXECVE_ERROR) {
-                mdebug1("Invalid path or permissions running command '%s'",command);
-                sprintf(*reason, "Invalid path or permissions running command '%s'",command);
-            } else if (result_code == 1) {
-                mdebug1("Timeout overtaken running command '%s'", command);
-                sprintf(*reason, "Timeout overtaken running command '%s'", command);
-            } else {
-                mdebug1("Internal error running command '%s'", command);
-                sprintf(*reason, "Internal error running command '%s'", command);
+    switch( wm_exec(command,&cmd_output,&result_code,data->commands_timeout,NULL)) {
+    case 0:
+        if (result_code > 0) {
+            mdebug1("Command (%s) returned code %d.", command, result_code);
+            if (*reason == NULL){
+                os_malloc(OS_MAXSTR, *reason);
+                if (result_code == EXECVE_ERROR) {
+                    mdebug1("Invalid path or permissions running command '%s'",command);
+                    sprintf(*reason, "Invalid path or permissions running command '%s'",command);
+                } else {
+                    mdebug1("Internal error running command '%s'", command);
+                    sprintf(*reason, "Internal error running command '%s'", command);
+                }
             }
+            os_free(cmd_output);
+            return 2;
+        }
+        break;
+    case 1:
+        if (*reason == NULL) {
+            os_malloc(OS_MAXSTR, *reason);
+            mdebug1("Timeout overtaken running command '%s'", command);
+            sprintf(*reason, "Timeout overtaken running command '%s'", command);
         }
         os_free(cmd_output);
         return 2;
-    } else if (result_code != 0) {
+    default:
```

Comment:
Default case have same debug messages and fill the reason field with the same message as the case 0 when `result_code < 0`.

Try to look for another _user friendly_ message for this use case, otherwise, we have no way to differentiate both cases when reading logs and check results.